### PR TITLE
🍰  feature/redis

### PR DESCRIPTION
**Pull request by Olderestin**
## What changed?     
Replaced sqlite with redis.
Update the whole project according to this task.

## Checklist before requesting a review     
- [x] I have performed a self-review of my code
- [x] I have run pre-commit over my code
- [x] Code is functional on my machine(Windows 10/Debian GNU/Linux 11 (bullseye))

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1 +1,5 @@
-BOT_TOKEN = ''
\ No newline at end of file
+BOT_TOKEN = ''
+
+REDIS_HOST = 'redis' # «redis» for Docker build, «localhost» for local usage
+REDIS_PORT = 6379
+REDIS_DB = 0
```
Comment (memhard) [inline .env.example]: consider not using all redis related data in env, because these parameters definitely cannot change

File: bot/db.py [hunk 2]
```diff
@@ -1,102 +1,59 @@
...
 
         Args:
-            param user_id: The ID of the user to be added.
+            redis_host: The hostname of the Redis server.
+            redis_port: The port number of the Redis server.
+            redis_db: The database index to use.
         """
-
-        self.cursor.execute('INSERT OR IGNORE INTO user (user_id) VALUES (?)', (user_id,))
+        self.redis = redis.StrictRedis(host=redis_host, port=redis_port, db=redis_db)
```
Comment (memhard) [inline bot/db.py:22]: in initilization of class add annotations to all variables!

File: bot/__main__.py [hunk 3]
```diff
@@ -1,30 +1,38 @@
...
     application.run_polling()
 
-if __name__ == '__main__':
+
+if __name__ == "__main__":
     main()
 
-logger.warning("App has been closed")
\ No newline at end of file
+logger.warning("App has been closed")
```
Comment (memhard) [inline bot/__main__.py:47]: unusable log

File: bot/__main__.py [hunk 4]
```diff
@@ -1,30 +1,38 @@
...
 
-def main() -> None:
-    
-    with USER_DB as user_db:
-        user_db.create_user_table()
 
+def main() -> None:
     application = ApplicationBuilder().token(settings.BOT_TOKEN).build()
 
     logger.info("Start the app")
```
Comment (memhard) [inline bot/__main__.py]: don't understand log, what is up, and it is started?

**Review by memhard — COMMENTED** [review]
LGTM 👍 but need changes + mypy

**Review by Olderestin — COMMENTED** [review]
Task done.

**Review by memhard — APPROVED** [review]
LGTM 👍 can approve

**Pull request merged**